### ISSUE TITLE
YAML incompatibilities between Ruby and Go

### ISSUE THREAD

**Issue by mikeantonelli** [newrelic/infrastructure-agent-chef#18]
There appears to be an incompatibility between Ruby's `YAML.dump` and `gopkg.in/yaml.v2` [used by the agent](https://github.com/newrelic/infra-integrations/blob/v1.0.0/vendor/vendor.json#L79-L84) to load configuration files. When configuring the MySQL on-host integration I would expect to utilize configuration for this cookbook that looks like:

```ruby
   host_integrations: {
      config: {
        mysql: {
          integration_name: 'com.newrelic.mysql',
          instances: [
            {
              name: 'mysql-status',
              command: 'status',
              arguments: {
                hostname: 'localhost',
                port: 3306,
                username: '...',
                password: '...',
                extended_metrics: 1,
                extended_innodb_metrics: 1
              },
              labels: {
                role: 'write-replica'
              }
            }
          ]
        }
      }
    }
```

The `mysql.yaml` written looks like this:

```yaml
---
integration_name: com.newrelic.mysql
instances:
- name: mysql-status
  command: status
  arguments:
    hostname: localhost
    port: 3306
    username: ...
    password: ...
    extended_metrics: 1
    extended_innodb_metrics: 1
  labels:
    role: write-replica
```

However, when the infrastructure agent attempts to read this file, I get `error=interface conversion: yaml.Node is yaml.Map, not yaml.List` (larger output below).

```
13:50:25 # newrelic-infra -config /etc/newrelic-infra/agent.yaml -debug
INFO[0000] Go pprof debugging enabled on http://localhost:6060
INFO[0000] New Relic Infrastructure Agent version 1.0.785 Creating Service (1.108174ms)
INFO[0000] Agent service manager started successfully. (1.168391ms)  service=newrelic-infra
INFO[0000] New Relic Infrastructure Agent version 1.0.785 Initializing (2.145651ms)
ERRO[0004] Error loading the plugin instance file        error=interface conversion: yaml.Node is yaml.Map, not yaml.List
INFO[0004] New Relic Infrastructure Agent version 1.0.785 Running (4.218323707s)
WARN[0004] unable to use SELinux                         error=SELinux status: disabled
^C2017/10/13 13:50:47 Orderly shutdown commenced
INFO[0006] Gracefully Exiting
INFO[0012] Agent service manager shutdown completed successfully.  service=newrelic-infra
```

If I manually update the YAML to prefix the `instances` stanza with two spaces (see below) it works fine:

```yaml
---
integration_name: com.newrelic.mysql
instances:
  - name: mysql-status
    command: status
    arguments:
      hostname: localhost
      port: 3306
      username: ...
      password: ...
      extended_metrics: 1
      extended_innodb_metrics: 1
    labels:
      role: write-replica
```

**Comment by taharah** [comment]
I logged this as an issue with New Relic as well since the agent is not correctly parsing YAML, but I am not sure where they are with getting the agent updated.

**Comment by smith** [comment]
Fixed in #19.
